apps/server/src/modules/provisioning/strategy/tsp/tsp.jwt.payload.ts

```diff
@@ -0,0 +1,35 @@
+import { IsString, IsOptional, IsArray } from 'class-validator';
+import { JwtPayload } from 'jsonwebtoken';
+
+export class TspJwtPayload implements JwtPayload {
+	@IsString()
+	public sub!: string;
+
+	@IsOptional()
+	@IsString()
+	public sid!: string;
+
+	@IsOptional()
+	@IsString()
+	public ptscListRolle!: string;
+
+	@IsOptional()
+	@IsString()
+	public personVorname!: string;
+
+	@IsOptional()
+	@IsString()
+	public personNachname!: string;
+
+	@IsOptional()
+	@IsString()
+	public ptscSchuleNummer!: string;
+
+	@IsOptional()
+	@IsArray()
+	public ptscListKlasseId!: [];

```

Comment:
If these properties are optional then their type definition should also indicate that they can be undifined like this:

```typescript
@IsOptional()
@IsString()
public sid: string | undifined;
```

Reply:
done